Makefile

```diff
@@ -61,8 +61,7 @@ test: run-unit-tests
 
 .PHONY: lint
 lint: build-slim build-test
-	# Note: Make sure that you run `git config diff.noprefix false` in this repo
-	# See this issue for more details: https://github.com/golangci/golangci-lint/issues/948
+	if [ $$(git config --get diff.noprefix) == "true" ]; then echo -e "\nLinting won't work because you have a git setting 'diff.noprefix' set to 'true',\nwhich does not work reliably with the golangci-lint --new-from-rev flag. \n\nTo run the linter, set 'diff.noprefix' setting to 'true' for this repository using the following command:\n\n'git config diff.noprefix false'\n\nSee https://github.com/golangci/golangci-lint/issues/948 for more details."; exit 1; fi
```

Comment:
This line is very difficult to add comments too :)

I'd do (but @johananl is the messages expert): s/Linting won't work/Linter can't run/

But I'm sure @johananl will think of a better way :)

Reply:
I think I actually wanted to write `Linting won't work reliably`. Would that make more sense? Because the linter can run, it just won't produce legitimate output.

Reply:
I don't think that is better, it seems confusing as I think it will work sometimes (i.e. not reliably, but sometimes).

I think something like this is better, but I'm okay with almost any text. It is clear enough in all versions:

`The linter has a bug and can't run with the current git config: 'git.diffprefix' is set to 'true'. To disable it, you can use the following command: 'git config diff.noprefix false'. See https://github.com/golangci/golangci-lint/issues/948 for more details.`

Reply:
Okay, I rephrased your suggestion a bit. This is how the output looks like right now:
```sh
$ make lint
CGO_ENABLED=0 GO111MODULE=on go build \
        -mod=vendor \
        -ldflags "-X github.com/kinvolk/lokomotive/pkg/version.Version=`git describe --tags --always`-dirty -extldflags '-static'" \
        -buildmode=exe \
        -o lokoctl \
        github.com/kinvolk/lokomotive/cmd/lokoctl
go test -run=nonexistent -mod=vendor -tags="aws,packet,aks,e2e,baremetal,disruptivee2e,poste2e" -covermode=atomic -buildmode=exe -v ./... > /dev/null
if [ $(git config --get diff.noprefix) == "true" ]; then printf "\n\nThe linter has a bug and can't run with the current git config: 'diff.noprefix' is set to 'true'. To override this setting for this repository,
run the following command:\n\n'git config diff.noprefix false'\n\nFor more details, see https://github.com/golangci/golangci-lint/issues/948.\n\n\n"; exit 1; fi


The linter has a bug and can't run with the current git config: 'diff.noprefix' is set to 'true'. To override this setting for this repository, run the following command:

'git config diff.noprefix false'

For more details, see https://github.com/golangci/golangci-lint/issues/948.


make: *** [Makefile:64: lint] Error 1
```

What do you think?

Reply:
The text looks fine for me :)